enhancements/network/networking-console-plugin.md

```diff
@@ -0,0 +1,148 @@
+---
+title: networking-console-plugin
+authors:
+  - "@upalatucci"
+reviewers:
+  - "@orenc1"
+  - "@zshi-redhat"
+  - "@pcbailey"
+  - "@metalice"
+  - "@spadgett"
+  - "@jhadvig"
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - None
+creation-date: 2024-04-12
+last-updated: 2024-04-12
+tracking-link:
+  - https://issues.redhat.com/browse/CONSOLE-3952
+---
+
+
+# Create console plugin for the networking section
+
+## Summary
+
+At present, several pages within the networking section of our console are defined within the console repository. 
+However, working with the console repository has slowed down development and there are multiple motivations to convert static console plugins into dynamic ones described [here](https://github.com/spadgett/enhancements/blob/master/enhancements/console/dynamic-plugins.md#motivation). 
```

Comment:
I am not a UI expert so could you please explain what is the difference between a static plugin and dynamic plugin?
Also when I read the provided link it states: "Static plugins can only be updated on the OpenShift release cadence. We want to enable plugin teams to move faster."
CNO also is the same.. we put in new features on release cadence, maybe I am missing something?

Reply:
this is the console repo: https://github.com/openshift/console
In this repo you can find:
- the main code that handle the console structure and provide an SDK for plugins
- static plugins that using the SDK can the integrated into the console and add pages and functionality ([example](https://github.com/openshift/console/tree/master/frontend/packages/network-attachment-definition-plugin))

Static plugins are built within the console. 

Dynamic plugins instead are defined in separate repo ([example](https://github.com/kubevirt-ui/kubevirt-plugin)) where they can define their workflow, dependenices and move faster in general. Those plugins CAN be built with a separate lifecycle and needs to be integrated by other operators to be part of openshift. 
The operator has to handle a couple of resources to make this work. 

The main benefit is not what you are mentioning but this:

`We already have a half-dozen teams contributing to console. We need to be able to scale as more teams contribute. This will be difficult if everyone is contributing to a single monorepo.`

Our experience with kubevirt-plugin proved that this works and we now move faster than ever.

Reply:
>Our experience with kubevirt-plugin proved that this works and we now move faster than ever.

Yes, this is the way how the console plugins should be delivered, by the matching operator. Console itself can be hosting any more of plugins, matter of fact is that we are gradually try to move the plugins, both static and dynamic, from our repo because of various reasons, some of them also mentioned here.